packages/server/api/src/app/ai/chat/ai-chat.service.ts

```diff
@@ -64,102 +81,128 @@ export const generateChatIdForMCP = (params: {
 
 export const createChatContext = async (
   chatId: string,
+  userId: string,
+  projectId: string,
   context: ChatContext | MCPChatContext,
 ): Promise<void> => {
   await cacheWrapper.setSerializedObject(
-    chatContextKey(chatId),
+    chatContextKey(chatId, userId, projectId),
     context,
     DEFAULT_EXPIRE_TIME,
   );
 };
 
 export const getChatContext = async (
   chatId: string,
+  userId: string,
+  projectId: string,
 ): Promise<ChatContext | null> => {
-  return cacheWrapper.getSerializedObject(chatContextKey(chatId));
+  return cacheWrapper.getSerializedObject(
+    chatContextKey(chatId, userId, projectId),
+  );
 };
 
 export const getChatHistory = async (
```

Comment:
I would extract all session management logic to a separate session management service.